qa/rpc-tests/remove_sprout_shielding.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+from decimal import Decimal
+from test_framework.authproxy import JSONRPCException
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises,
+    bitcoind_processes,
+    connect_nodes_bi,
+    initialize_chain,
+    start_nodes, start_node, get_coinbase_address,
+    wait_and_assert_operationid_status,
+    check_node_log, 
+    nuparams, BLOSSOM_BRANCH_ID, HEARTWOOD_BRANCH_ID, CANOPY_BRANCH_ID
+)
+
+import logging
+import time
+
+HAS_CANOPY = [nuparams(BLOSSOM_BRANCH_ID, 205), nuparams(HEARTWOOD_BRANCH_ID, 210), nuparams(CANOPY_BRANCH_ID, 220), '-nurejectoldversions=false']
+
+class RemoveSproutShieldingTest (BitcoinTestFramework):
+
+    def setup_chain(self):
+        print("Initializing test directory "+self.options.tmpdir)
+        initialize_chain(self.options.tmpdir)
+
+    def setup_nodes(self):
+        return start_nodes(4, self.options.tmpdir, extra_args=[
+            HAS_CANOPY,
+            HAS_CANOPY,
+            HAS_CANOPY,
+            HAS_CANOPY
+        ])
+
+    def run_test (self):
+
+        # Generate blocks up to Heartwood activation
+        logging.info("Generating initial blocks.")
+        self.nodes[0].generate(10)
+        self.sync_all()
+
+        # Shield coinbase to Sprout on node 0. Should pass
+        midAddr_0 = self.nodes[0].z_getnewaddress('sprout')
```

Comment:
Why is this called `midAddr_0` (in particular, the "mid")? It isn't clear to me.

Reply:
i've renamed this to `sprout_addr`.